[docs] Add expo-tools, testing, and PR guidance to CLAUDE.md (#46580)
expo/expo: .claude/CLAUDE.md

> ## expo-tools (`et`)

Reviewer: ```suggestion
## expotools (`et`)
```

it's expotools: https://github.com/expo/expo/blob/main/tools/bin/expotools.js

> ## expo-tools (`et`)

Reviewer 2: ```suggestion
## expotools (`et`)
```

> This repo ships its own CLI, `expo-tools`, available directly as `et <command> <...args>`.

Reviewer: note that `et` is only available when we run `direnv allow`

either doc to run `direnv allow` or use `node ./tools/bin/expotools.js` instead

> This repo ships its own CLI, `expo-tools`, available directly as `et <command> <...args>`.

Reviewer 2: ```suggestion
This repo ships its own CLI, `expotools`, available directly as `et <command> <...args>`.
```

Call it as `et`, not via `npx`, `bunx`, or `pnpm run` — it's already on the PATH. Many of our
processes (pod install, native unit tests, prebuild, and more) run through it.

Commands you'll reach for during normal development:

> - `et pod-install` — install pods in the directories that are out of sync. Run after changing

Reviewer: take a note that is required for ios/macos development only. we don't need this for android development

  dependencies or a podspec.

> - `et native-unit-tests` — run native unit tests for all packages that provide them. Scope to

Reviewer: move this under the native tests section?

  one package with `--packages <name>` (e.g. `et native-unit-tests -p ios --packages @expo/ui`).
- `et check-packages` — verify that packages build and their tests pass.

Run `et --help` to discover the rest (publishing, changelogs, on-call dashboards, and more).

## Rules

- **Red/green:** write the test first and watch it fail, then implement the feature or fix until
  it passes. Don't write the implementation before the failing test exists.

## Writing tests

### Unit tests

JS/TS unit tests use Jest and live next to the code in `__tests__/` (or `*.test.ts`). They run
as part of `et check-packages` (see Committing below).

### Native tests

Swift/Kotlin unit tests live in `packages/<pkg>/ios/Tests/` and `android/`, and run through
`et native-unit-tests` (scope with `--packages <name>`). Adding or changing an iOS `test_spec`

> requires `et pod-install` first.

Reviewer 2: Honestly, I think we shouldn't recommend using `et pod-install`. It's a bit fragile and slow as it also runs for Expo Go, not just bare-expo. It's better to just run it directly in `apps/*/{ios,macos}`, no matter of the app that you're working on.

## Committing

> Before committing, run `et check-packages`. It runs the unit tests **and** builds the package's

Reviewer 2: ```suggestion
Before committing, run `et check-packages <...packages>` with the package names that has changed. It runs the unit tests **and** builds the package's
```

Reviewer 2: Maybe we could add an option that runs checks for only uncommitted packages? `-u, --uncommitted`?

JS files. The build output is committed alongside your source changes, so a commit that skips
this step will be missing its built files. Stage both your source edits and the regenerated
build output.

## Creating PRs

Follow the contribution guide: https://github.com/expo/expo/blob/main/CONTRIBUTING.md

**Commit message:** format as `[platform][api] Title`, e.g. `[ios][video] Fix black screen on older devices`.

> **PR description** follows the repo template — fill in each section:

Reviewer: could link to the template https://github.com/expo/expo/blob/main/.github/PULL_REQUEST_TEMPLATE
